packages/headless/doc-parser/src/interface-resolver.test.ts

```diff
@@ -159,16 +160,124 @@ describe('#resolveInterfaceMembers', () => {
+    const docComment = buildMockApiDocComment(
+      '/**\n * Adds a callback that will be called on state change.\n *\n * @param listener - A callback to be invoked on state change.\n *\n * @returns An unsubscribe function to remove the listener.\n */\n'
+    );
+
+    const method = buildMockApiMethodSignature({
+      docComment,
+      excerptTokens: [
+        buildContentExcerptToken('subscribe(listener: '),
+        buildContentExcerptToken('() => void'),
+        buildContentExcerptToken('): '),
+        buildReferenceExcerptToken(
+          'Unsubscribe',
+          '@coveo/headless!Unsubscribe:interface'
+        ),
+        buildContentExcerptToken(';'),
+      ],
+      name: 'subscribe',
+      parameters: [
+        {
+          parameterName: 'listener',
+          parameterTypeTokenRange: {startIndex: 1, endIndex: 2},
+        },
+      ],
+      returnTypeTokenRange: {startIndex: 3, endIndex: 4},
+    });
+
+    const unsubscribeInterface = buildMockApiInterface({
+      name: 'Unsubscribe',
+    });
+
+    const unsubscribeCallSignature = buildMockApiCallSignature({
+      excerptTokens: [
+        buildContentExcerptToken('(): '),
+        buildContentExcerptToken('void'),
+      ],
+    });
+
+    controllerInterface.addMember(method);
+    unsubscribeInterface.addMember(unsubscribeCallSignature);
+
+    entryPoint.addMember(controllerInterface);
+    entryPoint.addMember(unsubscribeInterface);
+
+    const result = resolveInterfaceMembers(entryPoint, controllerInterface);
+
+    const paramEntity = buildMockEntity({
+      name: 'listener',
+      type: '() => void',
+      desc: 'A callback to be invoked on state change.',
+    });
+
+    const memberEntity = buildMockEntity({
+      name: '(call)',
+      type: '(): void',
+    });
+
+    const returnType = buildMockObjEntity({
+      name: 'returnType',
+      type: 'Unsubscribe',
+      typeName: 'Unsubscribe',
+      isOptional: false,
+      desc: 'An unsubscribe function to remove the listener.',
+      members: [memberEntity],
+    });
+
+    const funcEntity = buildMockFuncEntity({
+      name: 'subscribe',
+      desc: 'Adds a callback that will be called on state change.',
+      params: [paramEntity],
+      returnType: returnType,
+    });
+    expect(result).toEqual([funcEntity]);
+  });
+
+  it('resolves a call signature with primitive types', () => {
+    const entryPoint = buildMockEntryPoint();
+    const unsubscribeInterface = buildMockApiInterface({name: 'Unsubscribe'});
+
+    const method = buildMockApiCallSignature({
+      excerptTokens: [
+        buildContentExcerptToken('(): '),
+        buildContentExcerptToken('void'),
+      ],
+    });
+
+    unsubscribeInterface.addMember(method);
+    entryPoint.addMember(unsubscribeInterface);
+
+    const result = resolveInterfaceMembers(entryPoint, unsubscribeInterface);
+
+    const entity = buildMockEntity({
+      name: '(call)',
+      type: '(): void',
+    });
+    expect(result).toEqual([entity]);
```

Comment:
Shouldn't the resolved entity be a func entity? If the `Unsubscribe` function took a parameter, how would we represent it using a primitive entity?

Reply:
We should probably have a chat about this. `Unsubscribe` represents a function, yes, but it's an _interface_ representing a function type. In theory, we could have something that looks like this:

```
export interface IChooser {
  // A call signature:
  (x: number, y: number): number;

  // Another overload for this call signature:
  (x: string, y: string): string;
}
```
(I found this example by looking at the [ApiCallSignature docs](https://github.com/microsoft/rushstack.io-website/blob/master/pages/api/api-extractor-model.apicallsignature.md))

So we can't just store or display it like any other function, because it might have multiple call signatures. You can see on the [staging site](https://docsstaging.coveo.com/DOC-8502-generator/headless/latest/reference/controllers/facet/#unsubscribe) my first take on what it would look like.

At the moment, the changes I made allow us to parse and display any signature with any number of parameters, because it just takes all the tokens. However, this won't allow us to extract parameters that are complex. We might need another kind of entity in order to do this.